routes/api/contacts.js

```diff
@@ -1,25 +1,83 @@
-const express = require('express')
-
-const router = express.Router()
+const express = require('express');
+const Joi = require('joi');
+const router = express.Router();
+const contacts = require('../../models/contacts');
+const contactAddSchema = Joi.object({
+  name: Joi.string().required(),
+  email: Joi.string().required(),
+  phone: Joi.string().required(),
+})
 
 router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
+  try {
+    const result = await contacts.listContacts();
+    res.json(result);
+  } catch (error) {
+    next(error);
+  }
 })
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
+router.get('/:id', async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const result = await contacts.getContactById(id);
+    if (!result) {
+      res.status(404).json({ message: "Not found" });
+      return;
+    }
+    res.json(result);
+  } catch (error) {
+    next(error);
+  }
 })
 
 router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
+  try {
+    const { error } = contactAddSchema.validate(req.body);
+    if (error) {
+      res.status(400).json({ message: "missing required name field" });
+      return;
+    }
+    const result = await contacts.addContact(req.body);
+    res.status(201).json(result);
+  } catch (error) {
+    next(error);
+  }
 })
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
+router.put('/:id', async (req, res, next) => {
+  try {
+    const { error } = contactAddSchema.validate(req.body);
+    if (error) {
+      res.status(400).json({ message: "missing fields" });
+      return;
+    }
+    const { id } = req.params;
+    const result = await contacts.updateContactById(id, req.body);
+    if (!result) {
+      res.status(404).json({ message: "Not found" });
+      return;
+    }
+    res.json(result);
+  } catch (error) {
+    next(error);
+  }
 })
 
-router.put('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
+router.delete('/:id', async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const result = await contacts.removeContact(id);
+    if (!result) {
+      res.status(404).json({ message: "Not found" });
+      return;
```

Comment:
chyba po res nie ma sensu dawać returna?

Reply:
to już też wyjaśniłem, dlaczego tak jest